src/dex/uniswap-v3/uniswap-v3.ts

```diff
@@ -287,25 +287,29 @@ export class UniswapV3
     }
 
     this.logger.trace(`starting to listen to new pool: ${key}`);
+    const poolImplementation =
+      this.config.eventPoolImplementation !== undefined
+        ? this.config.eventPoolImplementation
+        : UniswapV3EventPool;
     pool =
       pool ||
-      new UniswapV3EventPool(
-        this.dexHelper,
-        this.dexKey,
-        this.stateMultiContract,
-        this.config.decodeStateMultiCallResultWithRelativeBitmaps,
-        this.erc20Interface,
-        this.config.factory,
-        fee,
-        token0,
-        token1,
-        this.logger,
-        this.cacheStateKey,
-        this.config.initHash,
+      new poolImplementation(
+          this.dexHelper,
+          this.dexKey,
+          this.stateMultiContract,
+          this.config.decodeStateMultiCallResultWithRelativeBitmaps,
+          this.erc20Interface,
+          this.config.factory,
+          fee,
+          token0,
+          token1,
+          this.logger,
+          this.cacheStateKey,
+          this.config.initHash,
       );
 
     try {
-      await pool.initialize(blockNumber, {
+      await pool!.initialize(blockNumber, {
```

Comment:
I don't get, why you need all these exclamation marks? I removed and it compiles 🤔

Reply:
@Verisana Strange, for some reasons I was getting ts errors.

Reply:
Do you still get errors or now compiles?

Reply:
@Verisana Removed exclamation marks, I don't know why for some reasons my IDE showed ts errors for these cases but now it's OK already.